From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fredrik=20S=C3=B6derquist?= <fs@opera.com>
Date: Mon, 26 Oct 2026 12:01:34 -0700
Subject: [M150] Make timed elements inserted into <use> shadow trees inert
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Original change's description:
> Make timed elements inserted into <use> shadow trees inert
>
> If timed elements somehow make it into a <use> shadow tree, they should
> not be scheduled on the timeline. Treat them the same as if they were
> not in the document at all.
>
> Fixed: 574791952
> Change-Id: I0a88c3a352139fbe2b93f4634794de929826ff47
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8610829
> Reviewed-by: Stephen Chenney <schenney@chromium.org>
> Commit-Queue: Stephen Chenney <schenney@chromium.org>
> Auto-Submit: Fredrik Söderquist <fs@opera.com>
> Cr-Commit-Position: refs/heads/main@{#1722305}

(cherry picked from commit 270d5d328addf47356621fb27f93873d5cd5e96f)

Bug: 575892588,574791952
Change-Id: I0a88c3a352139fbe2b93f4634794de929826ff47
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8659400
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Reviewed-by: Fredrik Söderquist <fs@opera.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/7871@{#4516}
Cr-Branched-From: f542126b8c1b3e80104b26bb05ec830bd1206f29-refs/heads/main@{#1639810}

diff --git a/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc b/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc
index b4b1a268b209908fdd5dbe34f7dceb1e800a6a0f..f2378f0fc3008b18b7cd76a8a52174bee2e1e2b4 100644
--- a/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc
+++ b/third_party/blink/renderer/core/svg/animation/svg_smil_element.cc
@@ -328,7 +328,9 @@ void SVGSMILElement::BuildPendingResource() {
   ClearResourceAndEventBaseReferences();
   DisconnectConditions();
 
-  if (!isConnected()) {
+  // Ignore timed elements that are inserted into a <use> shadow tree. This is
+  // currently possible via a cloned event handler.
+  if (!isConnected() || InUseShadowTree()) {
     // Reset the target element if we are no longer in the document.
     SetTargetElement(nullptr);
     return;
